--- kmediafactory-0.8.1/lib/kmftools.cpp.bero	2012-06-09 11:09:04.765197796 +0200
+++ kmediafactory-0.8.1/lib/kmftools.cpp	2012-06-09 11:09:23.865197892 +0200
@@ -51,6 +51,7 @@
 #include <fontconfig/fontconfig.h>
 #include <sys/stat.h>
 #include <stdlib.h>
+#include <unistd.h> // for rmdir
 
 #include "Q4puGenericSignalSpy.h"
 
--- kmediafactory-0.8.1/lib/videofile.cpp.bero	2012-06-09 11:09:36.659197954 +0200
+++ kmediafactory-0.8.1/lib/videofile.cpp	2012-06-09 11:11:42.801198567 +0200
@@ -58,7 +58,6 @@ bool VideoFile::open(const QString& file
     static bool init=false;
     if(!init) {
         av_register_all();
-        avcodec_init();
         avcodec_register_all();
         av_log_set_level(0);
         init=true;
@@ -66,7 +65,7 @@ bool VideoFile::open(const QString& file
 
     close();
 
-    if ( av_open_input_file(&avFromatContext, QFile::encodeName(filename).constData(), 0L, 0, 0L) != 0 ||
+    if ( avformat_open_input(&avFromatContext, QFile::encodeName(filename).constData(), 0L, 0L) != 0 ||
          av_find_stream_info(avFromatContext) < 0) {
         close();
     }
--- kmediafactory-0.8.1/plugins/video/videoobject.cpp.bero	2012-06-09 11:16:46.355200043 +0200
+++ kmediafactory-0.8.1/plugins/video/videoobject.cpp	2012-06-09 11:17:06.076200139 +0200
@@ -597,15 +597,15 @@ QVariant VideoObject::writeDvdAuthorXml(
             lang = "xx";
         }
 
-        QDomElement sub = doc.createElement("subpicture");
+        QDomElement sube = doc.createElement("subpicture");
 
         if ((lang == preferredLanguage) && (subFound == false)) {
             subFound = true;
             subTrack = i;
         }
 
-        sub.setAttribute("lang", lang);
-        titles.appendChild(sub);
+        sube.setAttribute("lang", lang);
+        titles.appendChild(sube);
         ++i;
     }
 
--- kmediafactory-0.8.1/src/CMakeLists.txt.bero	2012-06-09 11:14:44.763199451 +0200
+++ kmediafactory-0.8.1/src/CMakeLists.txt	2012-06-09 11:16:19.005199908 +0200
@@ -35,10 +35,14 @@ qt4_add_dbus_adaptor(kmediafactory_SRCS
 kde4_add_kcfg_files(kmediafactory_SRCS kmediafactorysettings.kcfgc)
 kde4_add_executable(kmediafactory ${kmediafactory_SRCS})
 if (KDE_4_4)
-target_link_libraries(kmediafactory kmf kmediafactoryinterfaces ${KDE4_KNEWSTUFF3_LIBS} ${QT_QTWEBKIT_LIBRARY})
+target_link_libraries(kmediafactory kmf kmediafactoryinterfaces ${KDE4_KNEWSTUFF3_LIBS} ${QT_QTWEBKIT_LIBRARY} ${KDE4_KIO_LIBS} ${KDE4_THREADWEAVER_LIBRARIES})
 else (KDE_4_4)
 target_link_libraries(kmediafactory kmf kmediafactoryinterfaces ${KDE4_KNEWSTUFF2_LIBS} ${QT_QTWEBKIT_LIBRARY})
 endif (KDE_4_4)
+if (UNIX)
+	# We use XDisplayString directly...
+	target_link_libraries(kmediafactory X11)
+endif(UNIX)
 
 install(TARGETS kmediafactory DESTINATION ${BIN_INSTALL_DIR})
 install(FILES kmediafactoryui.rc DESTINATION ${DATA_INSTALL_DIR}/kmediafactory)
