--- kmediafactory-0.8.1/plugins/output/kmediafactory_outputui.rc.bero	2012-06-09 12:46:34.274226244 +0200
+++ kmediafactory-0.8.1/plugins/output/kmediafactory_outputui.rc	2012-06-09 12:47:34.904226543 +0200
@@ -5,6 +5,8 @@
       <Separator/>
       <Action name="dvd_info" />
       <Action name="preview_dvd" />
+      <Action name="preview_dvd_mplayer" />
+      <Action name="preview_dvd_rosa_media_player" />
       <Action name="preview_dvd_xine" />
       <Action name="preview_dvd_kaffeine" />
       <Action name="preview_dvd_dragon" />
--- kmediafactory-0.8.1/plugins/output/outputplugin.cpp.bero	2012-06-09 12:47:45.403226589 +0200
+++ kmediafactory-0.8.1/plugins/output/outputplugin.cpp	2012-06-09 12:49:01.229226958 +0200
@@ -77,6 +77,8 @@ OutputPlugin::OutputPlugin(QObject *pare
     connect(dvdInfo, SIGNAL(triggered()), SLOT(slotDVDInfo()));
 #endif
 
+    addPreviewAction("mplayer", Qt::CTRL + Qt::Key_M, false);
+    addPreviewAction("rosa-media-player", Qt::CTRL + Qt::Key_R, false);
     addPreviewAction("xine", Qt::CTRL + Qt::Key_X);
     addPreviewAction("kaffeine", Qt::CTRL + Qt::Key_K);
     addPreviewAction("dragon", Qt::CTRL + Qt::Key_D, false, "dragonplayer");
