--- src/CMakeLists.txt.link	2012-07-30 04:56:10.000000000 +0000
+++ src/CMakeLists.txt	2012-07-30 04:57:19.000000000 +0000
@@ -35,9 +35,9 @@
 kde4_add_kcfg_files(kmediafactory_SRCS kmediafactorysettings.kcfgc)
 kde4_add_executable(kmediafactory ${kmediafactory_SRCS})
 if (KDE_4_4)
-target_link_libraries(kmediafactory kmf kmediafactoryinterfaces ${KDE4_KNEWSTUFF3_LIBS} ${QT_QTWEBKIT_LIBRARY})
+target_link_libraries(kmediafactory kmf kmediafactoryinterfaces ${KDE4_KNEWSTUFF3_LIBS} ${KDE4_KIO_LIBS} ${KDE4_THREADWEAVER_LIBRARIES} ${QT_QTWEBKIT_LIBRARY} ${X11_X11_LIB})
 else (KDE_4_4)
-target_link_libraries(kmediafactory kmf kmediafactoryinterfaces ${KDE4_KNEWSTUFF2_LIBS} ${QT_QTWEBKIT_LIBRARY})
+target_link_libraries(kmediafactory kmf kmediafactoryinterfaces ${KDE4_KNEWSTUFF2_LIBS} ${KDE4_KIO_LIBS} ${KDE4_THREADWEAVER_LIBRARIES} ${QT_QTWEBKIT_LIBRARY} ${X11_X11_LIB})
 endif (KDE_4_4)
 
 install(TARGETS kmediafactory DESTINATION ${BIN_INSTALL_DIR})
